### DISCUSSION TITLE
eMathHelp is in the ABP Filters english

### DISCUSSION THREAD

**Discussion by proy87** [abp-filters/abp-filters-anti-cv#768] [Q&A]
Hello.

I don't know why the site in the ABP filters and what exactly is blocked, but there are some problems with the code.
I'm using the plotly library.
<script src="https://cdn.plot.ly/plotly-2.8.3.min.js"></script>
<script>alert(Plotly)</script>
The above code is not working. Adblocker blocks the script src.

It appears that `TextDecoder` is blocked.

Any reason for this?

**Comment by manuel-martinez-dev** [comment]
Hi @proy87 

thanks for your message.

The site is known to push circumventing ads.

Can you provide an example what the script is supposed to do on the page?, what seems to be broken on the site?.

**Comment by proy87** [comment]
What is circumventing ads?
I will remove them and talk with publishers.
This page is not working: https://www.emathhelp.net/en/calculators/probability-statistics/box-and-whisker-plot-calculator/
The following library is used: https://cdn.plot.ly/plotly-2.8.3.min.js
And it uses TextDecoder.

**Reply by manuel-martinez-dev** [reply]
On the readme for the [repo](https://github.com/abp-filters/abp-filters-anti-cv) it's best explained :

_Some providers have begun to show ads that are harder to detect for ad-blocking tools like ABP, thus tricking ad-blocker users into downloading and displaying unrequested and intrusive advertising ("circumvention ads")._

so it's a data visualization library that the site is using, I see, I will update the filter list.

thanks for reporting this to us!

**Reply by proy87** [reply]
I'm the owner. 
Could you show me those ads on the site?
I will talk to publishers.
I respect the adblock users and against such ads. Just didn't know about their existence.

**Reply by manuel-martinez-dev** [reply]
I've updated the filters, the issue should not be reproducible anymore.

At the moment, I've not been able to replicate the ads, but in that the event they reappear, users will report them to us.

**Reply by proy87** [reply]
I don't want to bother you asking to notify me when users report a problem with emathhelp.
So can you show me an example of such ad on another site?

Thank you for your help.

**Reply by manuel-martinez-dev** [reply]
sure!,

you can go to the following site https://readarsenal.com/.

In this specific case, there are several ads that show up (banners).

![ads-redarsenal](https://user-images.githubusercontent.com/40897997/148427360-20462059-5094-4807-9863-51d418aae8ae.png)

**Comment by proy87** [comment]
It is just an iframe inside the div (as can be seen from the source code).
Is there any way to distinguish it from an "simple" ad?
How users detect that it is an circumvent ad?
